### PR TITLE
Upgrade Ruby/Rails and other dependencies

### PR DESCRIPTION
**Pull request by f1sherman**
I started out just trying to get specs passing on Ruby 2.7.0, which required upgrading Rails, but ended up upgrading several other gems as well.

Let me know if this is going overboard and I should break things up a bit.

### CODE REVIEW

File: spec/dummy/db/schema.rb [hunk 1]
```diff
@@ -11,128 +10,122 @@
...
     t.datetime "current_sign_in_at"
     t.datetime "last_sign_in_at"
-    t.string   "current_sign_in_ip",     limit: 255
-    t.string   "last_sign_in_ip",        limit: 255
+    t.string "current_sign_in_ip"
+    t.string "last_sign_in_ip"
     t.datetime "created_at"
     t.datetime "updated_at"
+    t.index ["email"], name: "index_admin_users_on_email", unique: true
+    t.index ["reset_password_token"], name: "index_admin_users_on_reset_password_token", unique: true
```
Comment (monkeyci) [inline spec/dummy/db/schema.rb]: Line is too long. [101/100]

File: spec/dummy/db/schema.rb [hunk 2]
```diff
@@ -11,128 +10,122 @@
...
+    t.text "body"
+    t.string "resource_id", null: false
+    t.string "resource_type", null: false
+    t.string "author_type"
+    t.integer "author_id"
     t.datetime "created_at"
     t.datetime "updated_at"
+    t.index ["author_type", "author_id"], name: "index_active_admin_comments_on_author_type_and_author_id"
+    t.index ["namespace"], name: "index_active_admin_comments_on_namespace"
+    t.index ["resource_type", "resource_id"], name: "index_active_admin_comments_on_resource_type_and_resource_id"
```
Comment (monkeyci) [inline spec/dummy/db/schema.rb]: Line is too long. [114/100]

File: spec/dummy/db/schema.rb [hunk 3]
```diff
@@ -11,128 +10,122 @@
...
-    t.string   "author_type",   limit: 255
+    t.string "namespace"
+    t.text "body"
+    t.string "resource_id", null: false
+    t.string "resource_type", null: false
+    t.string "author_type"
+    t.integer "author_id"
     t.datetime "created_at"
     t.datetime "updated_at"
+    t.index ["author_type", "author_id"], name: "index_active_admin_comments_on_author_type_and_author_id"
```
Comment (monkeyci) [inline spec/dummy/db/schema.rb]: Line is too long. [106/100]

**Comment by jgmontoya** [comment]
Hi @f1sherman, can you rebase to solve the conflicts so we can take a better look at this?

**Comment by f1sherman** [comment]
Done! Looks like I have a failing test now. When `value 1` is entered into the `select2` input focus is lost after `va` is typed. Any ideas what might have caused that?

image

**Comment by jgmontoya** [comment]
We've recently upgraded select2 to version 4.0.7, which apparently introduced this bug. It seems like this was solved on 4.0.8 with [PR #5576](https://github.com/select2/select2/pull/5576).

This issue should be solved as soon as #296 is merged.

**Comment by f1sherman** [comment]
Sounds good, thanks!

**Comment by jgmontoya** [comment]
#296 has been merged so it should work now :crossed_fingers: 

I would also like to get @rjherrera's feedback on this. I reckon the rails upgrade is breaking so maybe it would be more sensible to bundle this with #284 and other breaking changes for a version 2 release.

**Comment by f1sherman** [comment]
@jgmontoya sounds good, build is passing now!

**Pull request merged**